source/common/common/thread.h

```diff
@@ -47,8 +47,34 @@ class MutexBasicLockable : public BasicLockable {
   void unlock() UNLOCK_FUNCTION() override { mutex_.Unlock(); }
 
 private:
+  friend class CondVar;
   absl::Mutex mutex_;
 };
 
+/**
+ * Implementation of condvar, based on MutexLockable. This interface is a hybrid
+ * between std::condition_variable and absl::CondVar.
+ */
+class CondVar {
```

Comment:
There's not a lot to it, but does this class get adequate coverage from the existing tests?

Reply:
Yes, all lines covered, with many iterations.